backend/fichier/fichier.go

```diff
@@ -0,0 +1,387 @@
+package fichier
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ncw/rclone/fs"
+	"github.com/ncw/rclone/fs/config/configmap"
+	"github.com/ncw/rclone/fs/config/configstruct"
+	"github.com/ncw/rclone/fs/fshttp"
+	"github.com/ncw/rclone/fs/hash"
+	"github.com/ncw/rclone/lib/dircache"
+	"github.com/ncw/rclone/lib/pacer"
+	"github.com/ncw/rclone/lib/rest"
+	"github.com/pkg/errors"
+)
+
+const (
+	rootID     = "0"
+	apiBaseURL = "https://api.1fichier.com/v1"
+)
+
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "fichier",
+		Description: "1Fichier",
+		Config: func(name string, config configmap.Mapper) {
+		},
+		NewFs: NewFs,
+		Options: []fs.Option{
+			{
+				Help: "Your API Key, get it from https://1fichier.com/console/params.pl",
+				Name: "api_key",
+			},
+			{
+				Help:     "If you want to download a shared folder, add this parameter",
+				Name:     "shared_folder",
+				Required: false,
+				Advanced: true,
+			},
+		},
+	})
+}
+
+// Options defines the configuration for this backend
+type Options struct {
+	APIKey       string `config:"api_key"`
+	SharedFolder string `config:"shared_folder"`
+}
+
+// Fs is the interface a cloud storage system must provide
+type Fs struct {
+	root       string
+	name       string
+	features   *fs.Features
+	dirCache   *dircache.DirCache
+	baseClient *http.Client
+	options    *Options
+	pacer      *pacer.Pacer
+	rest       *rest.Client
+}
+
+// RoundTrip adds the Bearer token to every http request
+func (f *Fs) RoundTrip(request *http.Request) (response *http.Response, err error) {
```

Comment:
Functions which aren't used externally should be unexported so make this `roundTrip`.

That will stop people building code on functions we might want to change.